### PR TITLE
Add wolfSSL_DH_get0_pqg

### PR DESCRIPTION
**Pull request by TakayukiMatsuo**
Add wolfSSL_DH_get0_pqg.

### CODE REVIEW

File: src/ssl.c [hunk 1]
```diff
@@ -33577,9 +33577,31 @@ int wolfSSL_DH_set0_pqg(WOLFSSL_DH *dh, WOLFSSL_BIGNUM *p,
 
     return WOLFSSL_SUCCESS;
 }
+
 #endif /* v1.1.0 or later */
 #endif /* !HAVE_FIPS || HAVE_FIPS_VERSION > 2 */
 
+/**
+ * retrive p, q and g parameter
```
Comment (cconlon) [inline src/ssl.c:33585]: Spelling - "retrive" should be "Retrieve"

File: src/ssl.c [hunk 2]
```diff
@@ -33577,9 +33577,31 @@ int wolfSSL_DH_set0_pqg(WOLFSSL_DH *dh, WOLFSSL_BIGNUM *p,
...
+ * @param dh a pointer to WOLFSSL_DH
+ * @param p  a pointer to WOLFSSL_BIGNUM to be obtained dh
```
Comment (cconlon) [inline src/ssl.c:33587]: "obtained dh" should be "obtained from dh", also applies to next 2 lines below.

File: tests/api.c [hunk 3]
```diff
@@ -1617,7 +1617,63 @@ static void test_wolfSSL_CTX_SetTmpDH_buffer(void)
...
+static void test_wolfSSL_DH_get0_pqg(void)
+{
+#if defined(OPENSSL_EXTRA) && !defined(NO_DH)
+    DH *dh = NULL;
+    BIGNUM* p;
+    BIGNUM* q;
+    BIGNUM* g;
+
+#if defined(OPENSSL_ALL)
+#if !defined(HAVE_FIPS) || (defined(HAVE_FIPS_VERSION) && (HAVE_FIPS_VERSION>2))
```
Comment (cconlon) [inline tests/api.c:1629]: Minor style adjustment, please add space around ">" in "(HAVE_FIPS_VERSION > 2)"

File: tests/api.c [hunk 4]
```diff
@@ -1617,7 +1617,63 @@ static void test_wolfSSL_CTX_SetTmpDH_buffer(void)
...
+    FILE* f = NULL;
+    unsigned char buf[4096];
```
Comment (cconlon) [inline tests/api.c:1631]: Does "buf" need to be 4096 here?  The dh2048.der file is only 268 bytes.

**Review by cconlon — CHANGES_REQUESTED** [review]
Thanks, please see review comments.  Also, one of Hide's PRs was recently merged into master that added a stub for wolfSSL_DH_get0_pqg() in ssl.c.  May want to rebase this PR on master to pull in those changes.

File: src/ssl.c (hunk 1 above)
Comment (TakayukiMatsuo) [reply src/ssl.c]: By rebasing those comments are have been fixed.

File: src/ssl.c (hunk 2 above)
Comment (TakayukiMatsuo) [reply src/ssl.c]: Same above.

File: tests/api.c (hunk 3 above)
Comment (TakayukiMatsuo) [reply tests/api.c]: Adjusted.

File: tests/api.c (hunk 4 above)
Comment (TakayukiMatsuo) [reply tests/api.c]: Reduced to 268.

**Comment by TakayukiMatsuo** [comment]
Retrieved stub functions and prototypes by rebasing then moved the implementation to there.

**Review by cconlon — APPROVED** [review]

**Pull request merged**